Vindicta.Altis/Templates/initCategories.sqf

```diff
@@ -13,104 +13,137 @@ selectRandom (template_NATO select T_INF select T_INF_rifleman)
 OR it could be done through a get-function
 Author: Sparker 08.2017
 */
-
-T_SIZE = 8; //Number of categories in template
+#define T_DECLARE_CATEGORY(category, index, size) category = index; category##_SIZE = size; T_metadata set [index, [ #category , []]];
+#define T_DECLARE_ENTRY_OPT(category, id, index) id = index; (T_metadata select category select 1) set [index, [#id, []] ];
+#define T_DECLARE_ENTRY_REQ(category, id, index) id = index; (T_metadata select category select 1) set [index, [#id, []] ]; (T_metadata select category select 1 select index select 1) append
+T_metadata = []; //Category metadata
+
+// Faction classes
+T_FACTION_None = -1;
+T_FACTION_Civ = 0;
+T_FACTION_Guer = 1;
+T_FACTION_Military = 2;
+T_FACTION_Police = 3;
+
+T_SIZE = 9; //Number of categories in template
 T_WEIGHTS_OFFSET = T_SIZE; //Number of categories in template
 
-// = = = = = = = = = = I N F A N T R Y = = = = = = = = = = 
+T_metadata resize T_SIZE;
+
+fnc_get_T_metadata = {
```

Comment:
this file is not meant to put random functions in it, also bad function name, I will reorganize and rename it.

Reply:
general naming in SQF is like: module_fnc_doSomething, I like to stick to it when possible, also other template functions are named like `t_fnc_getDefaultCrew` and such, sorry to bother you much with this SQF nonsense (:

Reply:
Sure, I don't mind fixing this myself.